packages/lit-dev-content/site/tutorials/content/async-directive/06.md

```diff
@@ -0,0 +1,54 @@
+The thing we want to do in the timer callback is to update the value rendered by
+the directive -- but notice the timer callback fires _outside_ of the normal
+directive `update`/`render` lifecycle where we normally return a value for
+the directive to render.
+
+Here, we can use `AsyncDirective`'s `setValue` API to update the directive's
+value, outside of the directive's normal update cycle. To do this, we'll need to
+store the time passed to `update` in another class field:
+
+{% switchable-sample %}
+
+```ts
+  time!: Date;
+
+  update(part: Part, [time]: DirectiveParameters<this>) {
+    this.time = time;
+    ...
+  }
+```
+
+```js
+  update(part, [time]) {
+    this.time = time;
+    ...
+  }
+```
+
+{% endswitchable-sample %}
+
+This way we can reference the user's value passed in when the template was
+rendered from our timer callback.
+
+And then in our timer callback, we can call `setValue`, using our `render`
+result, which will re-format the date using the currently elapsed time.
+
+```ts
+        this.setValue(this.render(this.time));
```

Comment:
I think weirdness of `setValue` is highlighted here by the need to call `render`. If it has been previously explained, as suggested above, this should be fine.